subsys/bluetooth/host/Kconfig

```diff
@@ -70,19 +70,48 @@ config BT_HCI_RESERVE
 	  Headroom that the driver needs for sending and receiving buffers. Add a
 	  new 'default' entry for each new driver.
 
-config BT_RECV_IS_RX_THREAD
-	# Hidden option set by the HCI driver to indicate that there's
-	# no need for the host to have its own RX thread.
-	# If this option has been enabled it is then the responsibility of the
-	# HCI driver to call bt_recv_prio from a higher priority context than
-	# bt_recv in order to avoid deadlocks.
-	# If this option is disabled then only bt_recv should be called.
-	bool
-	prompt "bt_recv is called from RX thread" if BT_NO_DRIVER
+
+choice BT_RECV_CONTEXT
+	prompt "BT RX Thread Selection"
+	default BT_RECV_DIRECT if BT_LL_SW_SPLIT || BT_H4
+	default BT_RECV_THREAD
+	help
+	  Select the Bluetooth RX Thread.
+
+config BT_RECV_DIRECT
```

Comment:
Could we rename this to `BT_RECV_BLOCKING` and `BT_RECV_THREAD` to `BT_RECV_NONBLOCKING`, it's more obvious what's happening then.